library/modules/EventManager.cpp

```diff
@@ -455,18 +463,21 @@ static void manageJobStartedEvent(color_ostream& out) {
 
     static unordered_set<int32_t> startedJobs;
 
+    vector<df::job*> new_started_jobs;
     // iterate event handler callbacks
     multimap<Plugin*, EventHandler> copy(handlers[EventType::JOB_STARTED].begin(), handlers[EventType::JOB_STARTED].end());
     for (df::job_list_link* link = df::global::world->jobs.list.next; link != nullptr; link = link->next) {
         df::job* job = link->item;
         if (job && Job::getWorker(job) && !startedJobs.count(job->id)) {
             startedJobs.emplace(job->id);
-            for (auto &key_value : copy) {
-                auto &handler = key_value.second;
-                // the jobs must have a worker to start
-                DEBUG(log,out).print("calling handler for job started event\n");
-                handler.eventHandler(out, job);
-            }
+            new_started_jobs.emplace_back(job);
+        }
+    }
+    for (df::job* job : new_started_jobs) {
+        for (auto &[_,handle] : copy) {
+            // the jobs must have a worker to start
+            DEBUG(log,out).print("calling handler for job started event\n");
+            handle.eventHandler(out, job);
```

Comment:
this still suffers from the problem that if a handler cancels the job (which deletes it and removes it from the list), it will crash (probably).

to improve safety, it might be better to move the handler calls back inline and check after each handler call that the previous link still points to this job